modules/segger/Kconfig

```diff
@@ -82,6 +83,43 @@ config SEGGER_RTT_SECTION_DTCM
 config SEGGER_RTT_SECTION_CCM
 	bool "Place RTT data in the CCM linker section"
 
+if CPU_CORTEX_M
+
+config SEGGER_RTT_SECTION_CUSTOM
+	bool "Place RTT data in custom linker section at RAM start"
+
+config SEGGER_RTT_SECTION_CUSTOM_DTS_REGION
+	bool "Place RTT data in custom linker section defined by a memory region in DTS"
```

Comment:
I'm not fully familiar with devicetree & its use in Zephyr.

Where is the link with this config and the DTS configuration. Would you have an example for it please?

Reply:
Ping @giansta

Reply:
This feature was requested [here](https://github.com/zephyrproject-rtos/zephyr/pull/53569#discussion_r1247899876).
The link is just the config `SEGGER_RTT_SECTION_CUSTOM_NAME` that is enabled both using `SEGGER_RTT_SECTION_CUSTOM` and `SEGGER_RTT_SECTION_CUSTOM_DTS_REGION`. It's used in `SEGGER_RTT_Conf.h` in the modules repo.

Defining the DTS is then system or board specific. For my nRF52 board I tested it defining in DTS a region like this:
```
/ {
	soc {
		sram_rtt: memory@20000800 {
			compatible = "zephyr,memory-region", "mmio-sram";
			zephyr,memory-region = "SRAM_RTT";
			reg = <0x2000F800 0x800>;
		};
	};
};
```
The section name is defined in the line `zephyr,memory-region = "SRAM_RTT";` and both application and MCUboot prj.conf had to define `CONFIG_SEGGER_RTT_SECTION_CUSTOM_NAME="SRAM_RTT"`.